common/src/Microsoft.Azure.IIoT.Core/src/Diagnostics/TimeLogger.cs

```diff
@@ -0,0 +1,53 @@
+// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.Azure.IIoT.Services.Diagnostics {
+    using Microsoft.Azure.IIoT.Diagnostics;
+    using System;
+    using System.Diagnostics;
+
+    /// <summary>
+    /// Time execution
+    /// </summary>
+    public class TimeLogger : IDisposable {
+
+        /// <summary>
+        /// Create metric logger
+        /// </summary>
+        /// <param name="watchName"> Watch name eg. method name.</param>
+        /// <param name="metrics"> Metrics Logger.</param>
+        public TimeLogger(string watchName = "", IMetricLogger metrics = null) {
```

Comment:
metrics should never be passed as null - or else print will crash.  
Also, watchname to be empty string is probably also not desired.   Test both for != null.
Finally - to be more "fluent" would be nice to have an extension on IMetricLogger, so that: 
`
using (var timedOperation = _metrics.TrackDuration/TimeIt("op")) {
      // Long running operation
}
`

Reply:
Fluent approach is nicer :)